lang/syn/src/codegen/program/dispatch.rs

```diff
@@ -67,17 +67,38 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
                 #(#global_dispatch_arms)*
                 anchor_lang::idl::IDL_IX_TAG_LE => {
                     // If the method identifier is the IDL tag, then execute an IDL
-                    // instruction, injected into all Anchor programs.
-                    if cfg!(not(feature = "no-idl")) {
+                    // instruction, injected into all Anchor programs unless they have
+                    // no-idl enabled
+                    #[cfg(not(feature = "no-idl"))]
+                    {
                         __private::__idl::__idl_dispatch(
                             program_id,
                             accounts,
                             &ix_data,
                         )
-                    } else {
+                    }
+                    #[cfg(feature = "no-idl")]
+                    {
                         Err(anchor_lang::error::ErrorCode::IdlInstructionStub.into())
                     }
                 }
+                anchor_lang::event::EVENT_IX_TAG_LE => {
+                   // If the method identifier is the event tag, then execute an event cpi
+                   // against the noop instruction injected into all Anchor programs unless they have
+                   // no-cpi-events enabled.
+                    #[cfg(not(feature = "no-cpi-events"))]
```

Comment:
What would you say to making this feature opt-in instead of opt-out? I think it makes sense that IDL feature is opt-out since pretty much everyone uses Anchor IDLs but I don't think we can say the same for this feature.

Reply:
Yeah, I think it makes more sense to have this as `opt-in`. Do you want to make these changes?

Reply:
Sure, will do after reducing the repetition.

Reply:
Done in https://github.com/coral-xyz/anchor/pull/2438/commits/eb051efdb3ea24a376a4faa62e2f19efbc812ecc.